0001-typedb/02-dev/02-schema.md

```diff
@@ -0,0 +1,948 @@
+of a base type. Every type can have only one parent. There is also an internal type `role` that is used in relations. 
+Instances of types (i.e., the data) are known as `entities`, `relations`, and `attributes`.
+
+### Thing type
+
+There is a base type called `thing` type that can be used to address all types (both base and user-defined).
+
+<div class="note">
+[Warning]
+The `thing` base type will be deprecated in TypeDB version 3.0.
+</div>
+
+### Entity types
+
+Entity types define the objects within the business domain (e.g., an organization, file, or user). Further, an entity 
+(as in instance of an entity type e.g., Vaticle LTD) can exist independently of any other instance of any type 
+(e.g., John Doe).
+
+### Relation types
+
+Relation types define the roles participants can **play** in a relationship. The participants can be `entities`, 
+`relations`, or `attributes`. A `relation` type must specify at least one role. For example, `group-membership` 
+is a `relation` type that defines `user-group` and `group-member` roles. The `user-group` role is to be played 
+by a `user-group` entity whereas the `group-member` role is to be played by a `subject` entity.
+
+Roles allow the schema to enforce logical constraints. For example, a `group-membership` relation cannot associate 
+a `user` entity with a `file` entity.
+
+<div class="note">
+[Note]
+Roles can be queried like any other types even though roles do not have a direct type definition.
+</div>
+
+### Attribute types
+
+Attribute types represent the properties of `entity` and `relation` types (e.g., the name of a business unit or 
+a user) with a value. In TypeDB, any type can **own** an attribute type. However, different types can own the same 
+attribute type — and different instances can share ownership of the same attribute instance.
+
+For example, multiple users can own the same instance of an attribute type with the value of `Alex`.
+
+<div class="note">
+[Warning]
+The feature of attribute type owning another attribute types will be deprecated in TypeDB version 3.0.
+</div>
+
+### Inheritance
+
+A type can subtype another type. As a result, the subtype inherits all the attributes owned and roles played by its 
+supertype. However, while a type can only have a single supertype, types can be subtypes of other subtypes, 
+resulting in a type hierarchy. For example, `business unit` subtypes `user group`, which subtypes `subject`, 
+which subtypes `entity` base type.
+
+<div class="note">
+[Note]
+While roles are not generally considered to be types, they can be inherited and even overriden as a part of relation 
+inheritance.
+</div>
+
+## Rules
+
+Reasoning engine uses rules as a set of logic to infer new data. A rule consists of a condition and a conclusion. 
+Condition is a pattern to look for in data and conclusion is an action to take for every found pattern-matching piece 
```

Comment:
A conclusion is not an action to be performed. Rather, it contains data to be inserted.
I would expect an action to include inserts and deletes.

Reply:
A "found pattern-matching piece" should have a standard term such as "result" or conceptmap.
Whatever we use to refer to the result of a match.

Reply:
Fixed in https://github.com/vaticle/docs/commit/f81214c3303500311df1dc1ae6be223474eb6031#diff-be80f09573928c8d94fac728eebecdaa57b7679f8dab37db8f3d82b5edc04c0cR114